configure.ac

```diff
@@ -198,21 +198,36 @@ saved_user_CPPFLAGS="$CPPFLAGS"
 ##
 ##  Regex:
 ##
-##      -Werror         Literal -Werror
-##      \(              Start optional capturing group
-##      =               Literal equals sign
-##      [^[:space:]-]   Non-space characters
-##      \+              1 or more of the above
-##      \)              End optional capturing group
-##      \?              0 or 1 capturing group matches
+##      -Werror        Literal -Werror
+##      \(             Start optional capturing group
+##      =              Literal equals sign
+##      [^[:space:]]   Non-space characters
+##      \+             1 or more of the above
+##      \)             End optional capturing group
+##      \?             0 or 1 capturing group matches
 ##
-WERROR_SED= "sed -e 's/-Werror\(=[^[:space:]]\+\)\?//g'"
-
-CFLAGS="`echo $CFLAGS | $WERROR_SED`"
-CXXFLAGS="`echo $CXXFLAGS | $WERROR_SED`"
-FCFLAGS="`echo $FCFLAGS | $WERROR_SED`"
-JAVACFLAGS="`echo $JAVACFLAGS | $WERROR_SED`"
-CPPFLAGS="`echo $CPPFLAGS | $WERROR_SED`"
+## Note that the outer pair of '[]' ends up getting removed
+WERROR_SED='s/-Werror\(=[[^[:space:]]]\+\)\?//g'
```

Comment:
This was about the only reasonable way I could get this to work with the sed string inside a variable. Also, regardless of whether this is single or double-quoted, something appears to always be eating the outer pair of `[]`, so this adds an extra pair to be eaten. You can see this in any of the autotools runners, for example:

`/home/runner/work/hdf5/hdf5/configure: line 4807: sed -e 's/-Werror\(=^[:space:]\+\)\?//g': No such file or directory`

I also had to move the sed command out of this string to make things work.